### PR TITLE
[KHRGA-51] Updated DT vector types

### PR DESCRIPTION
**Pull request by imvucic**
Issue: KHRGA_51

### CODE REVIEW

File: source/iface/vector-types.rst [hunk 1]
```diff
@@ -8,6 +8,34 @@
...
 ************
 
+SYCL provides a cross-platform class template that works
+efficiently on SYCL devices as well as in host C++ code.
+This type allows sharing of vectors between the host and its
+SYCL devices. The vector supports member functions that allow
+construction of a new vector from a swizzled set of component
+elements.
+
+``vec<typename DataT``, int NumElements> is a vector type that compiles
```
Comment (mkinsner) [inline source/iface/vector-types.rst:18]:
```suggestion
``vec<typename DataT, int NumElements>`` is a vector type that compiles
```

File: source/iface/vector-types.rst [hunk 2]
```diff
@@ -8,6 +8,34 @@
...
+down to a SYCL backend built-in vector types on SYCL devices, where
```
Comment (mkinsner) [inline source/iface/vector-types.rst:19]:
```suggestion
down to a SYCL backend built-in vector type on SYCL devices, where
```

File: source/iface/vector-types.rst [hunk 3]
```diff
@@ -8,6 +8,34 @@
...
+possible, and provides compatible support on the host or when it is
```
Comment (mkinsner) [inline source/iface/vector-types.rst:20]:
```suggestion
possible, and provides compatible support on the host when 
```

Comment (mkinsner) [reply source/iface/vector-types.rst:20]: I've opened [this ](https://github.com/KhronosGroup/SYCL-Docs/pull/522) around these two proposed changes

File: source/iface/vector-types.rst [hunk 4]
```diff
@@ -68,241 +96,948 @@ Vector types
...
 
+   explicit constexpr vec(const DataT& arg);
+
+   template <typename... ArgTN> constexpr vec(const ArgTN&... args);
+
+   constexpr vec(const vec<DataT, NumElements>& rhs);
+
+   vec(vector_t nativeVector);
+
+Default construct a vector with element type ``DataT``
```
Comment (mkinsner) [inline source/iface/vector-types.rst:114]:
```suggestion
The first options default constructs a vector with element type ``DataT``
```

File: source/iface/vector-types.rst [hunk 5]
```diff
@@ -68,241 +96,948 @@ Vector types
...
+
+::
+
+   __swizzled_vec__ even() const
+
+Available only when: ``NumElements > 1``.
+
+Return an instance of the implementation-defined intermediate class
+template ``__swizzled_vec__`` representing an index sequence made
+up of the odd indexes of this SYCL ``vec`` which can be used to apply the
```
Comment (mkinsner) [inline source/iface/vector-types.rst:388]:
```suggestion
up of the even indexes of this SYCL ``vec`` which can be used to apply the
```

File: source/iface/vector-types.rst (hunk 3 above)
Comment (imvucic) [reply source/iface/vector-types.rst]: As I understand agreed removal of full first sentence in the second paragraph: 
_``vec<typename DataT, int NumElements>`` is a vector type that compiles
down to a SYCL backend built-in vector type on SYCL devices, where
possible, and provides compatible support on the host or when it is
not possible._ 
Done with commit : https://github.com/KhronosGroup/SYCL_Reference/pull/140/commits/10a0e11a4a1cfa16ba887d881ad24f7a806151b6

File: source/iface/vector-types.rst (hunk 1 above)
Comment (imvucic) [reply source/iface/vector-types.rst]: Fixed with : https://github.com/KhronosGroup/SYCL_Reference/pull/140/commits/10a0e11a4a1cfa16ba887d881ad24f7a806151b6

File: source/iface/vector-types.rst (hunk 4 above)
Comment (imvucic) [reply source/iface/vector-types.rst]: Done with https://github.com/KhronosGroup/SYCL_Reference/pull/140/commits/10a0e11a4a1cfa16ba887d881ad24f7a806151b6

File: source/iface/vector-types.rst (hunk 5 above)
Comment (imvucic) [reply source/iface/vector-types.rst]: Fixed with https://github.com/KhronosGroup/SYCL_Reference/pull/140/commits/10a0e11a4a1cfa16ba887d881ad24f7a806151b6

**Review by mkinsner — APPROVED** [review]

**Pull request merged**